tests/backup/backup_helper.go

```diff
@@ -3591,3 +3592,171 @@ func AdditionalScheduledBackupRequestParams(backupScheduleRequest *api.BackupSch
+
+	for _, userID := range userIDs {
+		userRuleOwnershipAccessConfig := &api.Ownership_AccessConfig{
+			Id:     userID,
+			Access: accessType,
+		}
+		userRuleOwnershipAccessConfigs = append(userRuleOwnershipAccessConfigs, userRuleOwnershipAccessConfig)
+	}
+
+	groupRuleOwnershipAccessConfigs := make([]*api.Ownership_AccessConfig, 0)
+
+	for _, groupID := range groupIDs {
+		groupRuleOwnershipAccessConfig := &api.Ownership_AccessConfig{
+			Id:     groupID,
+			Access: accessType,
+		}
+		groupRuleOwnershipAccessConfigs = append(groupRuleOwnershipAccessConfigs, groupRuleOwnershipAccessConfig)
+	}
+
+	ruleOwnershipUpdateReq := &api.RuleOwnershipUpdateRequest{
+		OrgId:     orgID,
+		Name:      ruleName,
+		Ownership: &api.Ownership{Groups: groupRuleOwnershipAccessConfigs, Collaborators: userRuleOwnershipAccessConfigs, Public: &publicAccess},
+		Uid:       ruleUid,
+	}
+
+	_, err := backupDriver.UpdateOwnershipRule(ctx, ruleOwnershipUpdateReq)
+	if err != nil {
+		return fmt.Errorf("failed to update rule ownership: %v", err)
+	}
+	return nil
+}
+
+// UpdateCloudCredentialOwnership Updates the CloudCredential object ownership
+func UpdateCloudCredentialOwnership(cloudCredentialName string, cloudCredentialUid string, userNames []string, groups []string, accessType api.Ownership_AccessType, publicAccess api.Ownership_PublicAccessControl, ctx context.Context) error {
+	log.Infof("UpdateCloudCredentialOwnership for users %v", userNames)
+	backupDriver := Inst().Backup
+	userIDs := make([]string, 0)
+	groupIDs := make([]string, 0)
+	for _, userName := range userNames {
+		userID, err := backup.FetchIDOfUser(userName)
+		if err != nil {
+			return err
+		}
+		userIDs = append(userIDs, userID)
+	}
+
+	for _, group := range groups {
+		groupID, err := backup.FetchIDOfGroup(group)
+		if err != nil {
+			return err
+		}
+		groupIDs = append(groupIDs, groupID)
+	}
+
+	userCloudCredentialOwnershipAccessConfigs := make([]*api.Ownership_AccessConfig, 0)
+
+	for _, userID := range userIDs {
+		userCloudCredentialOwnershipAccessConfig := &api.Ownership_AccessConfig{
+			Id:     userID,
+			Access: accessType,
+		}
+		userCloudCredentialOwnershipAccessConfigs = append(userCloudCredentialOwnershipAccessConfigs, userCloudCredentialOwnershipAccessConfig)
+	}
+
+	groupCloudCredentialOwnershipAccessConfigs := make([]*api.Ownership_AccessConfig, 0)
+
+	for _, groupID := range groupIDs {
+		groupCloudCredentialOwnershipAccessConfig := &api.Ownership_AccessConfig{
+			Id:     groupID,
+			Access: accessType,
+		}
+		groupCloudCredentialOwnershipAccessConfigs = append(groupCloudCredentialOwnershipAccessConfigs, groupCloudCredentialOwnershipAccessConfig)
+	}
+
+	cloudAccountOwnershipUpdateReq := &api.CloudCredentialOwnershipUpdateRequest{
+		OrgId:     orgID,
+		Name:      cloudCredentialName,
+		Ownership: &api.Ownership{Groups: groupCloudCredentialOwnershipAccessConfigs, Collaborators: userCloudCredentialOwnershipAccessConfigs, Public: &publicAccess},
```

Comment:
Can we split this up to increase readability?

```
&api.Ownership{
    Groups: groupCloudCredentialOwnershipAccessConfigs,
    Collaborators: userCloudCredentialOwnershipAccessConfigs,
    Public: &publicAccess,
}
```

Reply:
Done